src/dependency/Pyegi/Pyegi/main.py

```diff
@@ -131,63 +161,110 @@ def setupUi(self, MainWindow):
+            self.scripts_handler_ui.setupUi(self.scripts_handler_window, self)
+            self.scripts_handler_ui.update_feeds_process()
 
-    def retranslateUi(self, MainWindow):
+    def retranslateUi(self, window):
         _translate = QCoreApplication.translate
-        MainWindow.setWindowTitle(_translate("MainWindow", "MainWindow"))
+        window.setWindowTitle(_translate("MainWindow", "Main Window"))
         self.ScriptSelection_label.setText(_translate("MainWindow", "Select Script"))
         self.preview_label.setText(
             _translate("MainWindow", "No script selected to preview.")
         )
         self.cancel_pushButton.setText(_translate("MainWindow", "Cancel"))
         self.settings_pushButton.setText(_translate("MainWindow", "Settings"))
+        self.scripts_handler_pushButton.setText(
+            _translate("MainWindow", "Scripts Management")
+        )
         self.next_pushButton.setText(_translate("MainWindow", "Next"))
         self.LinesSelection_label.setText(_translate("MainWindow", "Apply script on:"))
         self.SelectedLines_radioButton.setText(
             _translate("MainWindow", "selected line(s)")
         )
         self.AllLines_radioButton.setText(_translate("MainWindow", "all lines"))
+        self.scriptSpecs_tabs.setTabText(
+            self.scriptSpecs_tabs.indexOf(self.preview_tab),
+            _translate("MainWindow", "Preview"),
+        )
+        self.scriptSpecs_tabs.setTabText(
+            self.scriptSpecs_tabs.indexOf(self.description_tab),
+            _translate("MainWindow", "Details"),
+        )
 
-    def openSettingsWindow(self, MainWindow):
-        window = QtWidgets.QMainWindow()
-        ui = Ui_SettingsWindow()
-        ui.setupUi(window, MainWindow)
-        window.show()
+    def openSettingsWindow(self):
+        self.child_window = QtWidgets.QMainWindow()
+        self.child_ui = Ui_SettingsWindow()
+        self.child_ui.setupUi(self.child_window, self)
+        self.child_window.show()
+
+    def openScriptsHandlerWindow(self):
+        self.combobox.setCurrentIndex(-1)
+        self.child_window = QtWidgets.QMainWindow()
+        self.child_ui = Ui_ScriptsHandlerWindow()
+        self.child_ui.setupUi(self.child_window, self)
+        self.child_window.show()
 
     def writeMainWindowOutput(self):
-        if self.selected_script == "":
+        if not self.selected_script:
             return
         main_py_parameters = {}
         if self.AllLines_radioButton.isChecked():
             main_py_parameters["applyOn"] = "all lines"
         else:
             main_py_parameters["applyOn"] = "selected lines"
-        main_py_parameters["selectedScript"] = self.selected_script
-        json.dump(main_py_parameters, open(system_inputs[1], "w"))
+        main_py_parameters["selectedScript"] = self.selected_script.id
+        with open(system_inputs[1], "w") as file:
+            json.dump(main_py_parameters, file, indent=4)
```

Comment:
just a thought: since these two lines are used often, isn't it better to define a function for them?